### DISCUSSION TITLE
X6  使用自定义vue节点 toJSON导出  使用formJSON   会出现一个beforeCreate 的报错

### DISCUSSION THREAD

**Discussion by patty-yang** [antvis/X6#1883] [Q&A]
求大佬们指点迷津

**Comment by NewByVector** [comment]
报错信息发一下呢，最好有个可复现的 codesanbox 链接。

**Reply by patty-yang** [reply]
https://codesandbox.io/s/falling-water-t696n6?file=/src/components/Customer.vue  

报错信息是  [Vue warn]: Error in beforeCreate hook: "TypeError: Cannot read properties of null (reading 'call')"

场景是  在toJSON之后 在另一个模块 使用from JSON出现的  大多数代码都在codesanbox 中

**Reply by NewByVector** [reply]
你序列化的姿势不太对，可以看下这个[文档](https://x6.antv.vision/zh/docs/tutorial/advanced/react#%E6%B8%B2%E6%9F%93-vue-%E8%8A%82%E7%82%B9)

**Comment by patty-yang** [comment]
已解决此问题
确实是序列化姿势不大对

感谢。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "antvis/X6"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月10日(星期四) 晚上8:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [antvis/X6] X6  使用自定义vue节点 toJSON导出  使用formJSON   会出现一个beforeCreate 的报错 (Discussion #1883)





 
你序列化的姿势不太对，可以看下这个文档
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by Miger0629** [reply]
你好，我也遇到了相同问题，toJSON导出，fromJSON导入，出现了同样的报错，但我研究了文档还是没有发现哪里存在问题（应该是我功力太差了），能请你详细描述你是如果解决问题的么 ？感谢@xiaoYangHAHAHA
